tests/test_git.py

```diff
@@ -1,12 +1,122 @@
 """This module contains the tests for the git container, the image with git pre-installed.
 """
-from bci_tester.data import GIT_CONTAINER
+import pytest
+from pytest_container.container import container_and_marks_from_pytest_param
+from pytest_container.container import DerivedContainer
+from pytest_container.pod import Pod
+from pytest_container.pod import PodData
 
+from bci_tester.data import GIT_CONTAINER
 
 CONTAINER_IMAGES = (GIT_CONTAINER,)
 
+PRIV_KEY = """-----BEGIN OPENSSH PRIVATE KEY-----
+b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAAAMwAAAAtzc2gtZW
+QyNTUxOQAAACDJMR7NW+gNZGborupz8XoZEjuKRuKjLzVPAwfPcjrTzQAAAIi/QgI2v0IC
+NgAAAAtzc2gtZWQyNTUxOQAAACDJMR7NW+gNZGborupz8XoZEjuKRuKjLzVPAwfPcjrTzQ
+AAAEBACrN2+98i3BPX40CQxih8gRePIokGrmrobXVnNja+XckxHs1b6A1kZuiu6nPxehkS
+O4pG4qMvNU8DB89yOtPNAAAAA0JDSQEC
+-----END OPENSSH PRIVATE KEY-----"""
+
+PUB_KEY = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIMkxHs1b6A1kZuiu6nPxehkSO4pG4qMvNU8DB89yOtPN BCI"
+
+GIT_SERVER_CONTAINERFILE = rf"""
+RUN zypper -n in git openssh-server
+RUN ssh-keygen -A
+RUN useradd -U -m -p "*" git
+
+RUN mkdir /home/git/.ssh && touch /home/git/.ssh/authorized_keys
+RUN chmod 700 /home/git/.ssh && chmod 600 /home/git/.ssh/authorized_keys
+RUN chown -R git:git /home/git/.ssh
+RUN echo "{PUB_KEY}" >> /home/git/.ssh/authorized_keys
+
+RUN mkdir -p /srv/git/project.git
+RUN git -C /srv/git/project.git init --bare
+
+RUN echo -e '\
+AuthorizedKeysFile      .ssh/authorized_keys\n\
+ChallengeResponseAuthentication no\n\
+PasswordAuthentication no\n\
+PermitRootLogin no\n\
+PubkeyAuthentication yes\n\
+AllowUsers git\n\
+UsePAM no\n\
+AllowTcpForwarding no\n\
+X11Forwarding no\n\
+Subsystem       sftp    /usr/lib/ssh/sftp-server\n\
+' > /etc/ssh/sshd_config
+
+EXPOSE 22
+
+CMD ["/usr/sbin/sshd", "-De"]
```

Comment:
The CI is still failing and I am not 100% sure why, but I suspect that the ssh daemon does not start quickly enough, try adding this:
```suggestion
CMD ["/usr/sbin/sshd", "-De"]

HEALTHCHECK --interval=5s --timeout=5s --retries=5 CMD ["/usr/bin/ssh-keyscan", "-H", "127.0.0.1"]
```

Reply:
I suspect the same. I can't find another reason, locally even with multiple runs it never fails.